metaflow/cmd/diff/__init__.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+import os
+import shutil
+import sys
+from subprocess import PIPE, run
+from tempfile import TemporaryDirectory
+
+from metaflow._vendor import click
+from metaflow import Run, namespace
+from metaflow.cli import echo_always
+
+EXCLUSIONS = [
+    "metaflow/",
+    "metaflow_extensions/",
+    "INFO",
+    "CONFIG_PARAMETERS",
+    "conda.manifest",
+]
+
+
+def echo(line):
+    echo_always(line, err=True, fg="magenta")
+
+
+def extract_code_package(runspec, exclusions):
+    try:
+        namespace(None)
+        run = Run(runspec)
+        echo(f"✅  Run *{runspec}* found, downloading code..")
+    except:
+        echo(f"❌  Run **{runspec}** not found")
+        sys.exit(1)
+
+    if run.code is None:
+        echo(
+            f"❌  Run **{runspec}** doesn't have a code package. Maybe it's a local run?"
+        )
+        sys.exit(1)
+
+    tar = run.code.tarball
+    members = [
+        m for m in tar.getmembers() if not any(m.name.startswith(x) for x in exclusions)
```

Comment:
minor nit: you technically want `startswith` if the exclusion ends in `/` but an exact match otherwise. I don't think it is hugely consequential but this could skip some files unintentionally.